### ISSUE TITLE
Fix S3459 FP: False positive with partial class and code generated constructor

### ISSUE THREAD

**Issue by nycdotnet** [SonarSource/sonar-dotnet#9674]
### Description

We use the NuGet package Autoconstructor in many of our solutions.  https://www.nuget.org/packages/AutoConstructor  With this package, if a class is marked `partial` and has the `[Autoconstructor]` attribute, this NuGet package will generate a constructor that accepts an argument for and will assign a value to any readonly non-static fields (with various customizations possible).

In Visual Studio, the Sonar NuGet analyzer is able to interpret the generated constructor, but not in CI where we are wiring in sonarcloud.io.  This means we will get S3459 wherever there are fields that are only assigned by the automatically generated constructor.  (We don't check in the generated code).

### Repro steps

```c#
namespace MyNameSpace;

[AutoConstructor]
public partial class SonarFalsePositive
{
    private TimeProvider TimeProvider { get; }

    public DateTimeOffset DoSomething()
    {
        return TimeProvider.GetUtcNow();
    }
}
```
### Expected behavior

No S3459 just like in Visual Studio.

### Actual behavior

S3459 is raised in SonarCloud.io:  `Remove unassigned auto-property 'TimeProvider', or set its value.`

### Known workarounds

We could disable S3459 in classes that use Autoconstructor.
I wonder if this rule could be disabled maybe for partial classes if it's expected that your CI product can't use the generated code?

### Related information

* Visual Studio 2022 17.11.4
* .NET 8
* SonarScanner NuGet 9.10.0.77988
* Windows 10 latest

**Comment by CristianAmbrosini** [comment]
I tried adding a reproducer but it's not an easy task. The AutoConstructorAttribute is auto-generated and this adds some complexity to our test infrastructure. I'll leave this open for the future, we might add a syntax-level check and avoid raising on classes marked with attributes that are called "AutoConstructor".

**Comment by nycdotnet** [comment]
OK thanks.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1541](https://sonarsource.atlassian.net/browse/NET-1541)

[NET-1541]: https://sonarsource.atlassian.net/browse/NET-1541?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
